cmd/kubeadm/app/cmd/join.go

```diff
@@ -219,6 +219,8 @@ func newCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 	joinRunner.AppendPhase(phases.NewControlPlanePreparePhase())
 	joinRunner.AppendPhase(phases.NewCheckEtcdPhase())
 	joinRunner.AppendPhase(phases.NewKubeletStartPhase())
+	joinRunner.AppendPhase(phases.NewControlPlaneJoinEtcdPhase())
+	joinRunner.AppendPhase(phases.NewKubeletWaitBootstrapPhase())
```

Comment:
I used the ControlPlaneJoin prefix to indicate that this is control-plane only, but fine to name it EtcdJoinPhase :-) should also be clear enough because there is no etcd on worker nodes 👍

Reply:
we are not consistent in naming these functions maybe we can go with `New-{ControlPlane}-Component-Action-SomethingElse-Phase`. and drop the "control plane" naming for etcd.

so i'm proposing NewEtcdJoinPhase (cmd `etcd-join`) and NewKubeletWaitBootstrapPhase (cmd `kubelet-wait-bootstrap`).

NewCheckEtcdPhase should be NewEtcdCheckPhase and cmd `etcd-check` but that's a hidden phase.